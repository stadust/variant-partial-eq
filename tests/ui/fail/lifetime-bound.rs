use std::marker::PhantomData;
use variant_partial_eq::VariantPartialEq;

#[derive(VariantPartialEq)]
struct MyStruct<'a, 'b: 'a> {
    ghost: PhantomData<(&'a str, &'b str)>
}

fn main() {}