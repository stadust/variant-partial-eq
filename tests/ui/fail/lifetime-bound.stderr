error: #[derive(VariantPartialEq)] does not support lifetime bounds
 --> tests/ui/fail/lifetime-bound.rs:5:25
  |
5 | struct MyStruct<'a, 'b: 'a> {
  |                         ^^
